pulp_ansible/app/tasks/collections.py

```diff
@@ -448,6 +455,43 @@ async def _fetch_collection_metadata(self, requirements_entry):
 
         await asyncio.gather(*tasks)
 
+    async def _read_from_downloaded_metadata(self, name, namespace, requirement):
+        tasks = []
+        loop = asyncio.get_event_loop()
+
+        collection_metadata = self._unpaginated_collection_metadata[namespace][name]
```

Comment:
Won't this key error if the requirements file specifies a collection that doesn't exist?